### PR TITLE
feat(prompt): finish prompt advanced keybindings

### PR DESCRIPTION
**Pull request by ngvt1n**
This PR: 
- implements `opts.position` for prompt to set intial cursor position 
- word navigation and word text editing for `utf8edit` and `prompt`

tracks #117

### CODE REVIEW

File: src/terminal/utf8edit.lua [hunk 1]
```diff
@@ -37,7 +37,11 @@ function UTF8EditLine:__tostring()
   return table.concat(res)
 end
 
+-- TODO: Should i setter/getter this? 👇
```
Comment (Tieske) [inline src/terminal/utf8edit.lua:40]: I think we should instantiate the object with an options table as well. This would just be the default value. So maybe rename to `default_word_delimiters`?

Then we can also add an option for the initial cursor position in this class as well.

File: spec/09-utf8edit_spec.lua [hunk 2]
```diff
@@ -826,4 +826,96 @@ describe("Utf8editLine:", function()
 
   end)
 
+
+
+  describe("left_word()", function()
+
+    it("moves the cursor to the start of the previous word (ASCII)", function()
```
Comment (Tieske) [inline spec/09-utf8edit_spec.lua:833]: can we add some edge cases to the tests?
- if cursor already at left
- if string is empty
- if there is only whitespace or separators to the left

similar for the other object methods

File: src/terminal/utf8edit.lua (hunk 1 above)
Comment (Tieske) [reply src/terminal/utf8edit.lua:40]: revisiting; the name can stay the same.

Also the we could accept a string or an options table for the initializer, such that just passing in a string is a shortcut to use all defaults.

File: spec/09-utf8edit_spec.lua (hunk 2 above)
Comment (ngvt1n) [reply spec/09-utf8edit_spec.lua:833]: Ok! I would say the true tester is the prompt though :P

**Review by Tieske — CHANGES_REQUESTED** [review]
this is some good stuff! ❤️

**Comment by ngvt1n** [comment]
> this is some good stuff! ❤️

Thaaaank you 😭😭😭
all done!
